traits/trait_types.py

```diff
@@ -2604,19 +2604,18 @@ def __init__(self, values, **metadata):
         default = self.default_value
         if 'default_value' in metadata:
             default = metadata.pop('default_value')
-            try:
-                default = self.validate(None, None, default)
-            except TraitError:
-                raise TraitError("Default value for PrefixTrait must be "
-                                 "a unique prefix present in the prefix list")
+            default = self.value_for(default)
         elif self.values:
             default = self.values[0]
 
         super().__init__(default, **metadata)
 
-    def validate(self, object, name, value):
+    def value_for(self, value):
         if not isinstance(value, str):
-            self.error(object, name, value)
+            raise TraitError(
+                "Value must be {}, but a value {!r} was specified.".format(
```

Comment:
Do we lose information about the object by doing this? What does the original error  message look like?

Reply:
No we won't. I expanded an existing test so we can be sure of it.

Reply:
Hmm now that I expanded it, I realized this sentence is missing the "of" and a mention of the value type.

Reply:
Sorry for the late changes, it is done.